src/parser.ts

```diff
@@ -1342,6 +1356,10 @@ export class Parser {
             expr = this.inheritCoverGrammar(this.matchKeyword('new') ? this.parseNewExpression : this.parsePrimaryExpression);
         }
 
+        if (isSuper && this.match('(') && !this.context.inClassConstructor) {
```

Comment:
@lahma:
this condition is not correct when the class is not inherited

Reply:
In esprima next there is also context.allowSuper. 
I got it from this pull req: https://github.com/jquery/esprima/pull/2047